src/codecs/png.rs

```diff
@@ -669,6 +681,11 @@ impl<W: Write> ImageEncoder for PngEncoder<W> {
             ))),
         }
     }
+
+    fn set_icc_profile(&mut self, icc_profile: Vec<u8>) -> Result<(), UnsupportedError> {
+        self.icc_profile = icc_profile;
```

Comment:
This is okay, but the png crate will need a fix: https://github.com/image-rs/image-png/issues/546